test/common/cdp.py

```diff
@@ -228,3 +230,10 @@ def set_frame(self, frame):
         self.cur_frame = frame
         if self.trace:
             print("-> switch to frame %s" % frame)
+
+    def read_console_line(self):
+        if not self.valid:
+            return
+
+        line = self._driver.stderr.readline()
+        return driver_line_re.match(line).group(2)
```

Comment:
This looks a bit fiddly. Also it cannot work as the `self._driver` process does not redirect its stderr. This is deliberate, as we want the user to see these messages from the driver.

If this is supposed to grab the first (or last) console message, then use `self.get_js_log()[0]` (or `[-1]` for the last one).

Reply:
> This looks a bit fiddly. Also it cannot work as the self._driver process does not redirect its stderr. This is deliberate, as we want the user to see these messages from the driver.

It does as of #8350. We need this for the unit tests, because the tap tests output control messages such as `cockpittest-tap-done`, which we need to catch in `tap-cdp`.

I see your point for wanting to see these messages for the integration tests, though. We could either add an `stderr` parameter to the cdp class or find a different way of communicating status from the qunit tests.